File: week6/pset/sentimental-mario-less/mario.py
def main():

    # calls height function for user input
    height = get_height()
    # calls print function to print pyramid
    p_print(height)


def get_height():
    # gets user input for height of pyramid. Input must be between 1 and 8
    while True:
        try:
            n = int(input("Height: "))
            if (1 <= n <= 8):
                return n
        except ValueError:
            print("enter a number between 1 and 8")


def p_print(height):

    # prints left justifyed pyramid incrementing through spaces and hash symbol printing.
    for row in range(height):
        for space in range(height - row - 1, 0, -1):
            print(" ", end="")
        for hash in range(row + 1):
            print("#", end="")
        print()


main()

File: week0/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int get_height(void);

int main(void)
{
    //get height of mario jump from 1 to 8
    int h = get_height();

    //print height of mario jump with spaces for blanks
    for (int row = 0; row < h; row++) //to print new line row
    {
        for (int space = h - row -1; space > 0; space--) //prints space
        {
            printf(" ");
        }
        for (int hash = 0; hash < row + 1; hash++)
        {
            printf("#");
        }
        printf("\n");
    }
}


int get_height(void)
{
     int h;
    do
    {
        h = get_int("Height: ");
    }
    while (h < 1 || h > 8);
    return h;
}